FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc

```diff
@@ -44,138 +46,95 @@ TrajectorySeedProducer::TrajectorySeedProducer(const edm::ParameterSet& conf):
-    simTrack_maxD0 = simTrackSelectionConfig.getParameter<double>("maxD0");
-    simTrack_maxZ0 = simTrackSelectionConfig.getParameter<double>("maxZ0");
-    //simtracks to skip (were processed in previous iterations)
-    std::vector<edm::InputTag> skipSimTrackTags = simTrackSelectionConfig.getParameter<std::vector<edm::InputTag> >("skipSimTrackIds");
-    
-    for ( unsigned int k=0; k<skipSimTrackTags.size(); ++k)
-    {
-        skipSimTrackIdTokens.push_back(consumes<std::vector<unsigned int> >(skipSimTrackTags[k]));
-    }
-
-
-    // The smallest number of hits for a track candidate
-    minLayersCrossed = conf.getParameter<unsigned int>("minLayersCrossed");
-
-    edm::InputTag beamSpotTag = conf.getParameter<edm::InputTag>("beamSpot");
-    if (beamSpotTag.label()!="")
-    {
-        testBeamspotCompatibility=true;
-        beamSpotToken = consumes<reco::BeamSpot>(beamSpotTag);
-    }
-    edm::InputTag primaryVertexTag = conf.getParameter<edm::InputTag>("primaryVertex");
-    if (primaryVertexTag.label()!="")
-    {
-        testPrimaryVertexCompatibility=true;
-        recoVertexToken=consumes<reco::VertexCollection>(primaryVertexTag);
-    }
-
-    //make sure that only one test is performed
-    if (testBeamspotCompatibility && testPrimaryVertexCompatibility)
+  // The name of the TrajectorySeed Collection
+  produces<TrajectorySeedCollection>();
+  const edm::ParameterSet& simTrackSelectionConfig = conf.getParameter<edm::ParameterSet>("simTrackSelection");
+  // The smallest pT,dxy,dz for a simtrack
+  simTrack_pTMin = simTrackSelectionConfig.getParameter<double>("pTMin");
+  simTrack_maxD0 = simTrackSelectionConfig.getParameter<double>("maxD0");
+  simTrack_maxZ0 = simTrackSelectionConfig.getParameter<double>("maxZ0");
+  //simtracks to skip (were processed in previous iterations)
+  std::vector<edm::InputTag> skipSimTrackTags = simTrackSelectionConfig.getParameter<std::vector<edm::InputTag> >("skipSimTrackIds");
+  
+  for ( unsigned int k=0; k<skipSimTrackTags.size(); ++k)
     {
-        throw cms::Exception("FastSimulation/Tracking/TrajectorySeedProducer: bad configuration","Either 'beamSpot' or 'primaryVertex' compatiblity should be configured; not both");
+      skipSimTrackIdTokens.push_back(consumes<std::vector<unsigned int> >(skipSimTrackTags[k]));
     }
-    
-    // The name of the hit producer
-    edm::InputTag recHitTag = conf.getParameter<edm::InputTag>("recHits");
-    recHitToken = consumes<SiTrackerGSMatchedRecHit2DCollection>(recHitTag);
-
-    // read Layers
-    std::vector<std::string> layerStringList = conf.getParameter<std::vector<std::string>>("layerList");
-    for(auto it=layerStringList.cbegin(); it < layerStringList.cend(); ++it) 
+  // The smallest number of hits for a track candidate
+  minLayersCrossed = conf.getParameter<unsigned int>("minLayersCrossed");
+  edm::InputTag beamSpotTag = conf.getParameter<edm::InputTag>("beamSpot");
```

Comment:
remove beamSpotTag
remove Vertex and BeamSpot from includes (.h an .cc file)
while on it, remove SimTrackIdProducer from includes